### PR TITLE
stmtsummary: add request-units info in statements_summary

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve?

Issue Number: ref #49318

Problem Summary:

### What changed and how does it work?
Add following 7 columns in `information_schema.statements_summary`:
```
  `MAX_REQUEST_UNIT_READ` double unsigned NOT NULL COMMENT 'Max read request-unit cost of these statements',
  `AVG_REQUEST_UNIT_READ` double unsigned NOT NULL COMMENT 'Average read request-unit cost of these statements',
  `MAX_REQUEST_UNIT_WRITE` double unsigned NOT NULL COMMENT 'Max write request-unit cost of these statements',
  `AVG_REQUEST_UNIT_WRITE` double unsigned NOT NULL COMMENT 'Average write request-unit cost of these statements',
  `MAX_QUEUED_RC_TIME` bigint(22) unsigned NOT NULL COMMENT 'Max time of waiting for available request-units',
  `AVG_QUEUED_RC_TIME` bigint(22) unsigned NOT NULL COMMENT 'Max time of waiting for available request-units',
  `RESOURCE_GROUP` varchar(64) DEFAULT NULL COMMENT 'Bind resource group name'
```
And add `resource_group_name` in statements_summary's aggration key, so after this change, the same sql executed by under different resource groups with be collected as different record.

Result example:
```
mysql> select DIGEST_TEXT, MAX_REQUEST_UNIT_READ, AVG_REQUEST_UNIT_READ, MAX_REQUEST_UNIT_WRITE, AVG_REQUEST_UNIT_WRITE, MAX_QUEUED_RC_TIME, AVG_QUEUED_RC_TIME, RESOURCE_GROUP from statements_summary where AVG_REQUEST_UNIT_READ > 0 limit 1\G;
*************************** 1. row ***************************
           DIGEST_TEXT: select `c` from `sbtest1` where `id` = ?
 MAX_REQUEST_UNIT_READ: 0.7367986354166667
 AVG_REQUEST_UNIT_READ: 0.5131357532322988
MAX_REQUEST_UNIT_WRITE: 0
AVG_REQUEST_UNIT_WRITE: 0
    MAX_QUEUED_RC_TIME: 212279751
    AVG_QUEUED_RC_TIME: 35929510
        RESOURCE_GROUP: rg1
1 row in set (0.02 sec)
```

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by nolouch — APPROVED** [review]
lgtm

**Review by nolouch — APPROVED** [review]
lgtm

File: pkg/executor/adapter.go [hunk 1]
```diff
@@ -1950,6 +1954,9 @@ func (a *ExecStmt) SummaryStmt(succ bool) {
 		Prepared:            a.isPreparedStmt,
 		KeyspaceName:        keyspaceName,
 		KeyspaceID:          keyspaceID,
+		RUDetail:            ruDetail,
+		// TODO: use sessVars.StmtCtx.ResourceGroupName
```
[the remark this answers is not in the document]
Comment (glorv) [reply pkg/executor/adapter.go]: fixed

**Comment by glorv** [comment]
@tangenta PTAL, thanks

**Comment by glorv** [comment]
@mornyx @tangenta  PTAL

File: go.mod [hunk 2]
```diff
@@ -103,7 +103,7 @@ require (
 	github.com/stretchr/testify v1.8.4
 	github.com/tdakkota/asciicheck v0.2.0
 	github.com/tiancaiamao/appdash v0.0.0-20181126055449-889f96f722a2
-	github.com/tikv/client-go/v2 v2.0.8-0.20231219052137-6f9ba8327b75
+	github.com/tikv/client-go/v2 v2.0.8-0.20231220023527-4ce1e45671b3
```
Comment (CabinfeverB) [inline go.mod:106]: The client-go version is ahead of this. pls merge master.

**Review by CabinfeverB — APPROVED** [review]
rest LGTM

File: go.mod (hunk 2 above)
Comment (glorv) [reply go.mod]: done

**Review by tangenta — APPROVED** [review]

**Pull request merged**